lib/routes/12379/route.ts

```diff
@@ -0,0 +1,82 @@
+import type { DataItem, Route } from '@/types';
+import got from '@/utils/got';
+import iconv from 'iconv-lite';
+import type { AlertItem } from './types';
+import { generateContent } from './content';
+
+const BASE_URL = 'http://www.12379.cn/data/alarm_list_all.html';
+
+const handler: Route['handler'] = async (context) => {
+    const filterWords = context.req.param('filter').split(',').sort();
+    const feedLink = `https://rsshub.app/12379/alert/${filterWords.join(',')}`;
+
+    // Fetch the index page
+    const { data: response } = await got(BASE_URL, {
+        responseType: 'buffer',
+    });
+    const originAlertItems: AlertItem[] = JSON.parse(iconv.decode(response, 'UTF-8')).alertData;
```

Comment:
UTF-8 charset does not require decoding again. You can simply access it through `response.alertData`